packages/flutter_tools/lib/src/custom_devices/custom_devices_config.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:meta/meta.dart';
+
+import '../base/config.dart';
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/platform.dart';
+import 'custom_device_config.dart';
+
+/// Represents the custom devices config file on disk which in turn
+/// contains a list of individual custom device configs.
+class CustomDevicesConfig {
+  /// Load a [CustomDevicesConfig] from a (possibly non-existent) location on disk.
+  ///
+  /// The config is loaded on construction. Any error while loading will be logged
+  /// but will not result in an exception being thrown. The file will not be deleted
+  /// when it's not valid JSON (which other configurations do) and will not
+  /// be implicitly created when it doesn't exist.
+  CustomDevicesConfig({
+    @required FileSystem fileSystem,
+    @required Logger logger,
+    @required Platform platform
+  }) : _fileSystem = fileSystem,
+       _config = Config(
+         _kCustomDevicesConfigName,
+         fileSystem: fileSystem,
+         logger: logger,
+         platform: platform,
+         deleteFileOnFormatException: false
+       );
+
+  CustomDevicesConfig.test({
```

Comment:
Make this `@visibleForTesting` if possible

Reply:
done